src/plugins/internalnotification/README.md

```diff
@@ -0,0 +1,37 @@
+- infos = Plugin for internal notification
+- infos/author = Thomas Wahringer <waht@libelektra.org>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides =
+- infos/recommends =
+- infos/placements = postgetstorage postcommit
+- infos/status = global libc unittest concept unfinished experimental nodoc
+- infos/metadata =
+- infos/description = Plugin for internal notification
+
+## Usage ##
+
+Allows programs to register integer variables that get automatically updated
+when the value of the key changes.
+
+Use as global plugin:
+$ sudo kdb global-mount internalnotification
+
+## Exported Methods ##
+
+This plugin exports the function `int elektraInternalnotificationRegisterInt (Plugin * handle, int * variable, Key * key)`.
+If the given key is contained in a KeySet on a kdbGet or kdbSet operation, its value is
+converted from string to integer and the registered variable is updated with the current value.
+This also works if the registered key is a cascading key.
+The function address is exported as `system/elektra/modules/internalnotification/exports/elektraInternalnotificationRegisterInt`.
+The plugin handle required for this plugin is exported as `system/elektra/modules/internalnotification/exports/handle`.
+Please note that the plugin API may change as this plugin is experimental.
+
+## Todo/Issues ##
+
+- Correct data storage for registered keys? elektraPlugin(Set|Get)Data?
+- Can we get rid of exporting the plugin handle?
```

Comment:
Yes. The idea was that the user calls an initialization function, this function registers itself in _KDB->globalPlugins. That is also the place where you can lookup the handle.

So the user links against your plugin, global mounting of it is not required then.